pylibftdi/driver.py

```diff
@@ -72,36 +73,43 @@ class Driver(object):
     """
 
     # prefer libftdi1 if available. Windows uses 'lib' prefix.
-    _lib_search = {
-        'libftdi': ('ftdi1', 'libftdi1', 'ftdi', 'libftdi'),
-        'libusb': ('usb-1.0', 'libusb-1.0')
+    _lib_search: dict[str, list[str]] = {
+        'libftdi': ['ftdi1', 'libftdi1', 'ftdi', 'libftdi'],
+        'libusb': ['usb-1.0', 'libusb-1.0']
     }
 
-    def __init__(self, libftdi_search=None, **kwargs):
+    def __init__(self, libftdi_search: Optional[str]=None, **kwargs: dict[str, Any]) -> None:
         """
         :param libftdi_search: force a particular version of libftdi to be used
             can specify either library name(s) or path(s)
         :type libftdi_search: string or sequence of strings
```

Comment:
Key question here is whether this alternate is worth preserving - you've effectively made the 'or sequence of strings' no longer valid (previously `_load_library` would take None, str, or List[str] - but AFAICT your intent is to have it only take `None` or `str`). The use-case for multiple strings is to allow a user to have multiple options (e.g. to support multiple OS platforms) but still have some restrictions (e.g. only listing possible library names for libftdi 0.x). The use case for a single string is when the user knows exactly the path / library name they are using (maybe they've built it themselves), and the use-case for None is that it 'just works' in many situations.

Previously the values of `self._lib_search` could be either str or List[str] and these were just transparently passed to `_load_library`, but I think for type consistency in the `_load_library` interface it would be good at this point to still accept `Optional[str]|List[str]` and if it's a `str` then wrap it in a singleton list, so `self._lib_search` is `Optional[List[str]]`

Reply:
Got it, I understand the use cases now. I think I was thrown off by "sequence" and wondered how that would be represented by downstream applications. I haven't addressed `_load_library` yet, but I'll fix that typing as well.